src/adapters/node.js

```diff
@@ -18,6 +18,9 @@
             if(args.url) {
                 args.url = args.url.replace(args.baseUrl, '');
             }
+            if(args.url === '') {
+              args.url = '/';
```

Comment:
If baseUrl = "http://www.example.com/"  and args.url == baseURL, then this will make args.url = "/", so that the full URL would have two trailing slashses ("http://www.example.com//").

Reply:
I don't think so. The request package should handle how to combine them.

Reply:
Maybe it should, but could you check and make sure it does?

Reply:
This code is tested. Without the slash, the request won't go through.

Reply:
So, then the package does not know how to combine them, and the code needs to be more careful about whether baseUrl ends in a trailing slash or not.  I would suggest removing any trailing slash from baseUrl.

Reply:
No, regardless of how the base url is, url should have some path.

Reply:
I was not saying that it shouldn't, but that you should adjust baseURL to work with url, in case the two are equal and baseUrl ends with /.  You are already changing url, and therefore it is your responsibility to make sure the baseUrl/url combination works.

Reply:
You don't change the baseUrl. It is supposed to be more stable across multiple calls to the server. You only change the url, which could be different for different resources. All it needs is to make sure that it has some value, regardless of what is in the baseUrl. Regardless of any changes to baseUrl, if the url is empty, it won't work. Regardless of what is in baseUrl, if url has something in it, it will work, The problem is url. The baseUrl has nothing to do with the problem. I don't want to change something that is not broken.

Reply:
In your comment above, you said, "The request won't go through".  That sounds broken.  Please see my first comment for what I was asking you to test would work.

Reply:
"This code is tested. Without the slash, the request won't go through."

The fix is to add slash to url. Changing the baseUrl does not change that.